reports-scheduler/src/main/java/com/amazon/opendistroforelasticsearch/reportsscheduler/metrics/MetricName.java

```diff
@@ -0,0 +1,99 @@
+    REPORT_DEFINITION_INFO_SYSTEM_ERROR("report_definition.info.system_error", true),
+
+    REPORT_DEFINITION_DELETE_TOTAL("report_definition.delete.total", true),
+    REPORT_DEFINITION_DELETE_INTERVAL_COUNT("report_definition.delete.count", true),
+    REPORT_DEFINITION_DELETE_USER_ERROR("report_definition.delete.user_error", true),
+    REPORT_DEFINITION_DELETE_SYSTEM_ERROR("report_definition.delete.system_error", true),
+
+    REPORT_DEFINITION_LIST_TOTAL("report_definition.list.total", true),
+    REPORT_DEFINITION_LIST_INTERVAL_COUNT("report_definition.list.count", true),
+    REPORT_DEFINITION_LIST_USER_ERROR("report_definition.list.user_error", true),
+    REPORT_DEFINITION_LIST_SYSTEM_ERROR("report_definition.list.system_error", true),
+
+    REPORT_INSTANCE_UPDATE_TOTAL("report_instance.update.total", true),
+    REPORT_INSTANCE_UPDATE_INTERVAL_COUNT("report_instance.update.count", true),
+    REPORT_INSTANCE_UPDATE_USER_ERROR("report_instance.update.user_error", true),
+    REPORT_INSTANCE_UPDATE_SYSTEM_ERROR("report_instance.update.system_error", true),
+
+    REPORT_INSTANCE_INFO_TOTAL("report_instance.info.total", true),
+    REPORT_INSTANCE_INFO_INTERVAL_COUNT("report_instance.info.count", true),
+    REPORT_INSTANCE_INFO_USER_ERROR("report_instance.info.user_error", true),
+    REPORT_INSTANCE_INFO_SYSTEM_ERROR("report_instance.info.system_error", true),
+
+    REPORT_INSTANCE_LIST_TOTAL("report_instance.list.total", true),
+    REPORT_INSTANCE_LIST_INTERVAL_COUNT("report_instance.list.count", true),
+    REPORT_INSTANCE_LIST_USER_ERROR("report_instance.list.user_error", true),
+    REPORT_INSTANCE_LIST_SYSTEM_ERROR("report_instance.list.system_error", true),
+
+    REPORT_FROM_DEFINITION_TOTAL("on_demand.create.total", true),
+    REPORT_FROM_DEFINITION_INTERVAL_COUNT("on_demand.create.count", true),
+    REPORT_FROM_DEFINITION_USER_ERROR("on_demand.create.user_error", true),
+    REPORT_FROM_DEFINITION_SYSTEM_ERROR("on_demand.create.system_error", true),
+
+    REPORT_FROM_DEFINITION_ID_TOTAL("on_demand_from_definition.create.total", true),
+    REPORT_FROM_DEFINITION_ID_INTERVAL_COUNT("on_demand_from_definition.create.count", true),
+    REPORT_FROM_DEFINITION_ID_USER_ERROR("on_demand_from_definition.create.user_error", true),
+    REPORT_FROM_DEFINITION_ID_SYSTEM_ERROR("on_demand_from_definition.create.system_error", true),
+
+    REPORT_PERMISSION_USER_ERROR("permission_user_error", true),
+
+    DEFAULT("default", true);
+
+    private String name;
+    private boolean numerical;
+
+    MetricName(String name, boolean numerical) {
+        this.name = name;
+        this.numerical = numerical;
+    }
+
+    public String getName() {
+        return name;
+    }
+
+    public boolean isNumerical() {
+        return numerical;
+    }
+
+    public static List<String> getNames() {
+        return Arrays.stream(MetricName.values()).map(v -> v.name).collect(Collectors.toList());
```

Comment:
since this does not change can it be stored as constant?

Reply:
Not needed, removed.